lib/Interpreter/Interpreter.cpp

```diff
@@ -356,95 +365,97 @@ namespace cling {
     // itself? One could use a macro (simillar to __dso_handle) to block
     // assignemnt and get around the mangling issue.
     const char* Linkage = LangOpts.CPlusPlus ? "extern \"C\"" : "";
-    if (!NoRuntime && !SyntaxOnly) {
+    if (!NoRuntime) {
       if (LangOpts.CPlusPlus) {
         Strm << "#include \"cling/Interpreter/RuntimeUniverse.h\"\n"
                 "namespace cling { class Interpreter; namespace runtime { "
-                "Interpreter* gCling=(Interpreter*)" << thisP << "; }}\n";
+                "Interpreter* gCling";
+        AssignIfValue(Strm, ThisP, "Interpreter*", " }}");
```

Comment:
Why don't you set it to `nullptr` if it's `nullptr`, even for syntax-only? Also, I find the version without the extra function call (i.e. the original code that spells out the `<<`) much simpler to read.